cmd/flux/bootstrap_git.go

```diff
@@ -164,7 +166,6 @@ func bootstrapGitCmdRun(cmd *cobra.Command, args []string) error {
 		// Configure repository URL to match auth config for sync.
 		repositoryURL.User = nil
 		repositoryURL.Scheme = "https"
-		repositoryURL.Host = repositoryURL.Hostname()
```

Comment:
I think this should stay because `bootstrapArgs.tokenAuth` take precedence. Why not make sure we don't loose the port number here?

Reply:
This proposal was the first approach we thought about, but we wanted to fix the root cause and for that is this PR, because the same could happens with ssh connections when not in the port 22 :)

Reply:
Fixed as we discussed :) 
https://github.com/fluxcd/flux2/pull/2216#discussion_r770030286